modules/system/controllers/Settings.php

```diff
@@ -169,10 +187,22 @@ protected function createModel($item)
     }
 
     /**
-     * Locates a setting item for a module or plugin
+     * Locates a setting item for a module or plugin.
+     *
+     * If none of the parameters are provided, they will be auto-guessed from the URL.
+     *
+     * @param string|null $author
+     * @param string|null $plugin
+     * @param string|null $code
+     *
+     * @return array
      */
-    protected function findSettingItem($author, $plugin, $code)
+    protected function findSettingItem($author = null, $plugin = null, $code = null)
     {
+        if (is_null($author) || is_null($plugin)) {
+            [$author, $plugin, $code] = $this->autoGuessSettingItem();
```

Comment:
What version of PHP does this auto expanding require?

Reply:
PHP 7.1 introduced it, so it should be right for L6.